akka-docs/src/main/paradox/typed/actors.md

```diff
@@ -271,6 +271,149 @@ Therefore after creating the Actor system with the `main` Actor’s
 `Behavior` the only thing we do is await its termination.
 
 
+## Class based Actors behaviors
+
+The samples shown so far are all based on a functional programming style 
+where you pass a function to a factory which then constructs a behavior, for stateful 
+actors this means passing immutable state around as parameters and switching to a new behavior 
+whenever you need to act on a changed state. We recommend that you default to using 
+this style for defining actors as it makes the behaviors easier to reason about, easier 
+to compose with other behaviors and requires less boilerplate.
+
+In some cases there might still be a use for something that is closer to the untyped style of 
```

Comment:
"closer to an object oriented style where a concrete class for the actor behavior is defined
 and mutable state is kept inside of it as fields." ?

instead of comparing with untyped, since it's not less typed?